CMSgov/easi-app
# EASI-1868 Help Report A Problem

# EASI-1867

Design
https://www.figma.com/file/jrc18ENtS3ufj7ttr5vHKW/EASi-Basic-Page-Templates?node-id=27%3A4859

This work is similar to the Help / Send Feedback form https://github.com/CMSgov/easi-app/pull/1713

## Changes and Description

This Report A Problem form is modified from Send Feedback.

Some functions from Send Feedback are modified to be general enough for this version.

Simplified form-related constants and types into the same constants file.

### Dev notes
- Resorted to radio inputs each with data-testid
  Could not figure out how to target radio inputs with rtl's preferred query methods, with the current dom.
  Most examples i've found have unique text for radio labels.
  In this case there are repeating radio inputs that are distinguished by fieldset legends.
- Understand the need for apollo gql mockprovider, but is there any way to not have to define inputs in the mock query? Can it all somehow be driven by the rtl user events? The events fill in the same values for the form fields.




## How to test this change

Enable the `helpFooter` ld flag

- Jest
  - src/validations/helpSchema.test.ts
    Form field schema validations
  - src/views/Help/SendFeedback/index.test.tsx
    Form component error handling and submission
- Local dev http://localhost:3000/help/report-a-problem
  - Or alternatively click "Report a problem" towards the page footer
  - Qa the form and see a successful submit or field level errors
  - After the submission is complete, attempt to close the tab or start another form
